test/Datadog.Trace.Tests/HeadersCollectionTests.cs

```diff
@@ -36,6 +38,75 @@ public void WebRequest_InjectExtract_Identity()
             Assert.Equal(context.Origin, resultContext.Origin);
         }
 
+        [Fact]
+        public void WebRequest_Extract_EmptyHeadersReturnsNull()
+        {
+            IHeadersCollection headers = WebRequest.CreateHttp("http://localhost").Headers.Wrap();
+            var resultContext = SpanContextPropagator.Instance.Extract(headers);
+
+            Assert.Null(resultContext);
+        }
+
+        [Fact]
+        public void NameValueCollection_InjectExtract_Identity()
+        {
+            const int traceId = 9;
+            const int spanId = 7;
+            const SamplingPriority samplingPriority = SamplingPriority.UserKeep;
+            const string origin = "synthetics";
+
+            IHeadersCollection headers = new NameValueCollection().Wrap();
+            var context = new SpanContext(traceId, spanId, samplingPriority, null, origin);
+
+            SpanContextPropagator.Instance.Inject(context, headers);
+            var resultContext = SpanContextPropagator.Instance.Extract(headers);
+
+            Assert.NotNull(resultContext);
+            Assert.Equal(context.SpanId, resultContext.SpanId);
+            Assert.Equal(context.TraceId, resultContext.TraceId);
+            Assert.Equal(context.SamplingPriority, resultContext.SamplingPriority);
+            Assert.Equal(context.Origin, resultContext.Origin);
+        }
+
+        [Fact]
+        public void NameValueCollection_Extract_EmptyHeadersReturnsNull()
```

Comment:
We run the same tests with different implementations of IHeadersCollection. We could aggregate them into a single test by using [Theory] and [MemberData]

Reply:
Yep good point. What do you think of the new changes introduced in 7f5dece10b4a8dca97b60967a2ce716515751d55?

Reply:
Looking good!